src/treeView/items/activityTreeItem.ts

```diff
@@ -106,8 +114,25 @@ class ExerciseTreeItem extends AbstractActivityTreeItem {
                 "..",
                 "..",
                 "assets",
-                `exercise-${status}.svg`,
-            );
+                `exercise-${status}.svg`);
+        } else if (exercise.programming_language == null) {
+            this.iconPath = path.join(
+                __filename,
+                "..",
+                "..",
+                "..",
+                "..",
+                "assets",
+                `language-text.svg`);
+        } else {
+            this.iconPath = path.join(
+                __filename,
+                "..",
+                "..",
+                "..",
+                "..",
+                "assets",
+                `language-${String(exercise.programming_language?.name)}.svg`);
```

Comment:
I believe this should work. `?.` returns undefined if the value to the left was null, and `first || second` should return `second` if the value to the left is falsy. Avoids about 10 lines of code duplication. Also, is the `String()`-conversion required here? `programming_language.name` is already a `string` as far as I can see.
```suggestion
        } else {
            this.iconPath = path.join(
                __filename,
                "..",
                "..",
                "..",
                "..",
                "assets",
                `language-${String(exercise.programming_language?.name || "text")}.svg`);
```

Reply:
Thank you, I tried the shorter syntax and it works.